File: 2023/day_04/2.py
#!/usr/bin/env python3

cards = [] 

with open('./input.txt') as file:
    lines = file.readlines()
    sum = 0
    for idx in range(len(lines)):
        lines[idx] = lines[idx].rstrip("\n")
        win, got = lines[idx].split(":")[1].split("|")
        win = list(filter(lambda x: x != "", win.split(" ")))
        got = list(filter(lambda x: x != "", got.split(" ")))
        cards.append(
            list(
                range(
                    idx + 1, 
                    idx + 1 +len(set(win).intersection(set(got)))
                )
            )
        )

def sumcards(cardlist):
    sum = len(cardlist)
    for got in cardlist:
        sum += sumcards(cards[got])
    return sum

sum = len(cards)
for card in cards:
    sum += sumcards(card)
print(sum)

File: 2024/day_03/2.jl
#!/usr/bin/env julia 

lines = []

open("./input.txt") do file
  for line in eachline(file)
    push!(lines,line)
  end
end

text = join(lines)

validparts = split(text,r"don't\(\).*?do\(\)")
println(sum(map(x -> sum([prod(map(y -> parse(Int,y), match.captures)) for match in eachmatch(r"mul\((\d+),(\d+)\)", x)]), validparts)))

